src/Build/Evaluation/Evaluator.cs

```diff
@@ -2076,6 +2102,17 @@ private void EvaluateImportElement(string directoryOfImportingFile, ProjectImpor
                 PerformDepthFirstPass(importedProjectRootElement);
             }
 
+            if (!importedAnything && hasCondition)
```

Comment:
```
  <Import Project="$(EmptyProperty)"/>
  <Import Project="$(EmptyProperty)" Condition="1 == 1"/>
```
These two are invalid and will cause a build error.

But the glob scenario is interesting, I will log that.